CMakeLists.txt

```diff
@@ -229,21 +229,22 @@ endif()
 #============================================================================
 # Example binaries
 #============================================================================
-
-add_executable(example test/example.c)
-target_link_libraries(example zlib)
-add_test(example example)
-
-add_executable(minigzip test/minigzip.c)
-target_link_libraries(minigzip zlib)
-
-if(HAVE_OFF64_T)
-    add_executable(example64 test/example.c)
-    target_link_libraries(example64 zlib)
-    set_target_properties(example64 PROPERTIES COMPILE_FLAGS "-D_FILE_OFFSET_BITS=64")
-    add_test(example64 example64)
-
-    add_executable(minigzip64 test/minigzip.c)
-    target_link_libraries(minigzip64 zlib)
-    set_target_properties(minigzip64 PROPERTIES COMPILE_FLAGS "-D_FILE_OFFSET_BITS=64")
+if(BUILD_TESTING)
```

Comment:
I recommend using:
```cmake
if(NOT DEFINED BUILD_TESTING OR BUILD_TESTING)
```
This way the behavior is backwards compatible. Also might want to consider adding an `option()`.

Reply:
I thought about adding a `option()` but it might not work to be backwards compatible because it expects a default value.

Reply:
When I look at what other projects are doing, I don't see anyone using `if(NOT DEFINED BUILD_TESTING OR BUILD_TESTING)`.  They all use `if(BUILD_TESTING)`.  Among the third-party dependencies of my project, I see this in both curl and glog.  abseil goes further with a custom option that is required in addition to `BUILD_TESTING`: `option(ABSL_BUILD_TESTING ...` and `if(BUILD_TESTING AND ABSL_BUILD_TESTING)`.

Furthermore, CTest docs show exactly what I've done here: https://cmake.org/cmake/help/latest/module/CTest.html

> Configure a project for testing with CTest/CDash
>
> Include this module in the top CMakeLists.txt file of a project to enable testing with CTest and dashboard submissions to CDash:
>
> ```
> project(MyProject)
> ...
> include(CTest)
> ```
>
> The module automatically creates a BUILD_TESTING option that selects whether to enable testing support (ON by default). After including the module, use code like:
>
> ```
> if(BUILD_TESTING)
>   # ... CMake code to create tests ...
> endif()
> ```
>
> to creating tests when testing is enabled.

So I think I will leave it as-is, if that's acceptable to the maintainers.  @nmoinvaz, do you have rights to approve PRs in this repo, or do I need to get the attention of @madler?  Thanks!

Reply:
The reason I recommended `if(NOT DEFINED BUILD_TESTING OR BUILD_TESTING)` was for backwards compatibility. There may be somebody using CMake in their CI or build system that depends on the inclusion of the `example` or `minigzip` projects by default. By using `if(BUILD_TESTING)` it forces all those people to update their systems with `-D BUILD_TESTING=ON`.  If zlib was a new project, then `if(BUILD_TESTING)` would be best but it is an old project with many people relying on it, and even relying on the CMake as it is.

Reply:
I do not have any extra rights here in this repository. It has to be approved by @madler and merged by him.

Reply:
It might be good idea to rebase on top of latest develop.

Reply:
I'll retarget this PR against the develop branch.